server/src/internalClusterTest/java/org/elasticsearch/index/TimeSeriesModeIT.java

```diff
@@ -0,0 +1,598 @@
+                    }
+                    mappings.endObject();
+                }
+                mappings.endObject();
+            }
+            mappings.endObject();
+        }
+        mappings.endObject();
+
+        String index = randomAlphaOfLength(5).toLowerCase(Locale.ROOT);
+        prepareCreate(index).setSettings(s).setMapping(mappings).get();
+        ensureGreen(index);
+
+        IndexResponse indexResponse = index(
+            index,
+            XContentFactory.jsonBuilder()
+                .startObject()
+                .field("foo", "bar")
+                .field("@timestamp", System.currentTimeMillis())
+                .field("new_field", "value")
+                .endObject()
+        );
+        assertEquals(indexResponse.getResult(), Result.CREATED);
+
+        XContentBuilder expect = XContentFactory.jsonBuilder();
+        expect.startObject();
+        {
+            expect.startObject("_doc");
+            {
+                expect.startObject(DataStreamTimestampFieldMapper.NAME);
+                {
+                    expect.field("enabled", true);
+                }
+                expect.endObject();
+                expect.startObject("properties");
+                {
+                    expect.startObject(DataStreamTimestampFieldMapper.DEFAULT_PATH);
+                    {
+                        expect.field("type", "date");
+                        expect.startObject("meta");
+                        {
+                            expect.field("field_meta", "time_series");
+                        }
+                        expect.endObject();
+                    }
+                    expect.endObject();
+                    expect.startObject("foo");
+                    {
+                        expect.field("type", "keyword");
+                        expect.field("time_series_dimension", true);
+                    }
+                    expect.endObject();
+                    expect.startObject("new_field");
+                    {
+                        expect.field("type", "text");
+                        expect.startObject("fields");
+                        {
+                            expect.startObject("keyword");
+                            {
+                                expect.field("type", "keyword");
+                                expect.field("ignore_above", 256);
+                            }
+                            expect.endObject();
+                        }
+                        expect.endObject();
+                    }
+                    expect.endObject();
+                }
+                expect.endObject();
+            }
+            expect.endObject();
+        }
+        expect.endObject();
+        GetMappingsResponse getMappingsResponse = client().admin().indices().prepareGetMappings(index).get();
+        assertThat(getMappingsResponse.getMappings().get(index).source().string(), equalTo(Strings.toString(expect)));
+    }
+
+}
```

Comment:
I'd tend to want to do these as yaml rest tests instead of `ESIntegTestCase`. The yaml tests are much more copy and paste-ish, we run them during rolling upgrade testing and during backwards compatibility testing.

I'm not sure that it's worth rewriting it now though.

Reply:
fine, I move the test from TimeSeriesModeTests to TimeSeriesModeIT, some tests are the same with yaml tests. Maybe later, we can remove these tests.